Explain how to change default text (#1306)
Altinn/altinn-studio-docs: content/app/development/ux/texts/_index.en.md

![Altinn Studio Repos](edit-texts-in-repos.png "Change texts in Altinn Studio Repos")

## Change dafault texts and error messages in an application

It is possible to change default texts and error messages that is displayed in the app. 
Here are the id:s with its default value in [English](https://github.com/Altinn/app-frontend-react/blob/v3/src/language/texts/en.ts), 

> [Norwegian Bokmål](https://github.com/Altinn/app-frontend-react/blob/v3/src/language/texts/nb.ts) and [Nynorsk](https://github.com/Altinn/app-frontend-react/blob/v3/src/language/texts/nn.ts) 

Reviewer: I think we could safely link to the main branch here, as v3 is on its way out.

Also, not sure which notation `id:s` is, but they could probably be called `keys` or just `IDs`. Look here for more: https://english.stackexchange.com/questions/212040/is-the-plural-form-of-id-spelled-ids-or-id

> Those default texts that has a number needs to be handled in a different way. For exampel `file_uploader_validation_error` displays an error message if 
> the user tries to continiue to the next page even though it is required to add one attachment. This default error message 
> will be displayed "You need to upload 1 attachement(s) to continue'.

Reviewer: ```suggestion
Those default texts that has a number needs to be handled in a different way. For example `file_uploader_validation_error` displays an error message if it is required to add one attachment. This default error message will be displayed as "You need to upload 1 attachement(s) to continue'.
```

The feature that stops the user from navigating to the next page is unrelated to this, and it can be shown without the app having that feature enabled, so it would't be fitting to mix that in here.

![Tekster i appen](defaultErrorMessageEN.png "Default text displayed in application")

These default text is devided in two strings, one before 
the number: `You need to upload` and one text resource after the number: `attachement(s) to continue`. The number is for the 

> moment not possible to edit since it is connected to max and min number of attachements, in this case. but the text surrounding the number is possible to change.

Reviewer: It might be worth it to note that we don't normally consider changing default texts in the app a _breaking change_, unless it is one of the few keys we've already [documented that you can change](https://docs.altinn.studio/app/development/configuration/process/customize/). So although it's nice to have documentation for how one can customize these things in an app, it also restricts future changes - for example such that we cannot really change this behavior in a rolling app-frontend update now that it's documented how you can overwrite the keys with your own text. I guess that's the direction we're trending towards anyway, but it might be nice to add a disclaimer here on this page noting that keys we use in app-frontend may change without warning. If you're relying heavily on rewriting default texts in an app you probably also want to [pin the app-frontend version number to some version you have tested against](https://docs.altinn.studio/nb/app/maintainance/dependencies/#app-frontend). That also leads to an increased burden on your part as an application developer, as you'll want to re-test and upgrade from time to time.

Author: If I've understood you correctly you do not think it is desirable to have this in the documentation? @olemartinorg
